src/gtnh/assembler/generic_assembler.py

```diff
@@ -0,0 +1,191 @@
+        config_version:GTNHVersion
+
+        modpack_config, config_version = self.get_config()
+        config_file: Path = get_asset_version_cache_location(modpack_config, config_version)
+
+        with ZipFile(config_file, "r", compression=ZIP_DEFLATED) as config_zip:
+            return len([item for item in config_zip.namelist() if item not in self.exclusions[side]])
+
+    def get_mods(self, side: Side) -> List[Tuple[GTNHModInfo | ExternalModInfo, GTNHVersion]]:
+        """
+        Method to grab the mod info objects as well as their targetted version.
+
+        :param side: the targetted side
+        :return: a list of couples where the first object is the mod info object, the second is the targetted version.
+        """
+        get_mod: Callable[
+            [str, str, Set[Side], ModSource], Optional[tuple[Union[GTNHModInfo, ExternalModInfo], GTNHVersion]]
+        ] = self.modpack_manager.assets.get_mod_and_version
+        valid_sides: Set[Side] = {side, Side.BOTH}
+
+        github_mods: List[Optional[Tuple[GTNHModInfo | ExternalModInfo, GTNHVersion]]] = [
+            get_mod(name, version, valid_sides, ModSource.github) for name, version in self.release.github_mods.items()
+        ]
+
+        external_mods: List[Optional[Tuple[GTNHModInfo | ExternalModInfo, GTNHVersion]]] = [
+            get_mod(name, version, valid_sides, ModSource.github)
+            for name, version in self.release.external_mods.items()
+        ]
+
+        mods: List[Tuple[GTNHModInfo | ExternalModInfo, GTNHVersion]] = list(filter(None, github_mods + external_mods))
+        return mods
+
+    def get_config(self) -> Tuple[GTNHConfig, GTNHVersion]:
+        """
+        Method to get the config file from the release.
+
+        :return: a tuple with the GTNHConfig and GTNHVersion of the release's config
+        """
+
+        config: GTNHConfig = self.modpack_manager.assets.config
+        version: Optional[GTNHVersion] = config.get_version(self.release.config)
+        assert version
+        return config, version
+
+    def add_mods(
+            self, side: Side, mods: list[tuple[GTNHModInfo, GTNHVersion]], archive: ZipFile, verbose: bool = False
+    ) -> None:
+        """
+        Method to add mods in the zip archive.
+
+        :param side: target side
+        :param mods: target mods
+        :param archive: archive being built
+        :param verbose: flag to turn on verbose mode
+        :return: None
+        """
+        pass
+
+    def add_config(
+            self, side: Side, config: Tuple[GTNHConfig, GTNHVersion], archive: ZipFile, verbose: bool = False
+    ) -> None:
+        """
+        Method to add config in the zip archive.
+
+        :param side: target side
+        :param config: a tuple giving the config object and the version object of the config
+        :param archive: archive being built
+        :param verbose: flag to turn on verbose mode
+        :return: None
+        """
+        pass
```

Comment:
if it is indeed an ABC you would annotate these, have these throw NotImplementedError, and expect the children to implement them.